src/system/SystemLayer.h

```diff
@@ -166,6 +173,16 @@ class DLL_EXPORT Layer
         return ScheduleLambdaBridge(std::move(bridge));
     }
 
+    typedef enum
+    {
+        APP_TASK = 0,
+        THREAD_TASK,
+        CHIP_TASK,
+        TASK_UNDEFINED
+    } TaskLocks_e;
+    virtual CHIP_ERROR LockTask(TaskLocks_e task) { return CHIP_NO_ERROR; }
+    virtual CHIP_ERROR UnlockTask(TaskLocks_e task) { return CHIP_NO_ERROR; }
```

Comment:
We cannot introduce an idea of locking at this layer, especially since it seems to expect split into multiple tasks which is not true on some platforms.

Please do not make this generic and follow the previous comment above to add interlockin of thread task to the ThreadStackManager